### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by sverhoeven**
Refs #221 , #219, #218

**List of related issues or pull requests**

Refs: #ISSUE_NUMBER

**Describe the changes made in this pull request**

**Instructions to review the pull request**

### CODE REVIEW

**Comment by sverhoeven** [comment]
Over 80% coverage, stopping

File: tests/github/fair_software/badge/test_checker_with_noreason_user_config.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+
+def get_mocked_checker(user_config_filename):
+    repo = Repo("https://github.com/fair-software/badge")
+    return Checker(repo, user_config_filename=user_config_filename)
+
+
+@pytest.fixture
+def user_config(tmp_path):
+    config = tmp_path / "howfairis.yml"
```
Comment (jspaaks) [inline tests/github/fair_software/badge/test_checker_with_noreason_user_config.py:15]: Be advised PR #225 has a more convenient way for reading mocked user files

File: tests/contracts/cli.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    @abstractmethod
+    def test_version_option(self, invoke_cli):
+        pass
+
+    @abstractmethod
+    def test_show_default_config(self, invoke_cli):
+        pass
+
+    @abstractmethod
+    def test_with_an_url(self, invoke_cli):
```
Comment (jspaaks) [inline tests/contracts/cli.py:15]:
```suggestion
    def test_with_a_url(self, invoke_cli):
```

File: tests/github/fair_software/badge/test_checker_with_noreason_user_config.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+skip_license_checks_reason: ""
+skip_registry_checks_reason: ""
+skip_citation_checks_reason: ""
+skip_checklist_checks_reason: ""
+"""
+    config.write_text(text)
+    yield config
+
+
+class TestCheckerWithReasonableUserConfig(Contract):
```
Comment (jspaaks) [inline tests/github/fair_software/badge/test_checker_with_noreason_user_config.py:26]: BTW How would you feel about using `TestChecker` in all of these tests? In my experience, the class name here tends to be out of sync with the containing file name.

File: tests/github/fair_software/badge/test_checker_with_noreason_user_config.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    def test_check_checklist(self, mocker: Mocker, user_config, capsys):
+        with mocker:
+            checker = get_mocked_checker(user_config)
+            assert checker.check_checklist() is True
+            captured = capsys.readouterr()
+            assert "no reason provided" in captured.out
```
Comment (jspaaks) [inline tests/github/fair_software/badge/test_checker_with_noreason_user_config.py:32]: I like this, nice and simple

File: tests/github/fair_software/badge/test_checker_with_noreason_user_config.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        pass
+
+    def test_is_on_github_marketplace(self, mocker: Mocker):
+        pass
+
+    def test_readme(self, mocker: Mocker):
+        pass
+
+    def test_repo(self, mocker: Mocker):
+        pass
```
Comment (jspaaks) [inline tests/github/fair_software/badge/test_checker_with_noreason_user_config.py:123]: the low-level check functions are not implemented. Was this intentional?

File: tests/github/fair_software/badge/test_checker_with_reasonable_user_config.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+def user_config(tmp_path):
+    config = tmp_path / "howfairis.yml"
+    text = """skip_repository_checks_reason: dunno1
+skip_license_checks_reason: dunno2
+skip_registry_checks_reason: dunno3
+skip_citation_checks_reason: dunno4
+skip_checklist_checks_reason: dunno5
+"""
+    config.write_text(text)
+    yield config
```
Comment (jspaaks) [inline tests/github/fair_software/badge/test_checker_with_reasonable_user_config.py:23]: Probably this fixture can go in `conftest.py`. While this introduces magic, it is the standard pattern

File: tests/github/fair_software/badge/test_cli_no_args.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      × has_zenodo_metadata_file
+(5/5) checklist:
+      × has_core_infrastructures_badge
+
+Calculated compliance: ● ● ● ● ○
+
+It seems you have not yet added the fair-software.eu badge to your README.md. You can do so by pasting the following snippet:
+
+[![fair-software.eu](https://img.shields.io/badge/fair--software.eu-%E2%97%8F%20%20%E2%97%8F%20%20%E2%97%8F%20%20%E2%97%8F%20%20%E2%97%8B-yellow)](https://fair-software.eu)
+"""
```
Comment (jspaaks) [inline tests/github/fair_software/badge/test_cli_no_args.py:72]: Perhaps we need another directory next to PR #225's `user-files` and `repo-files` that can contain these snippets

File: tests/github/fair_software/doesnotexist/test_repo_no_args.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import pytest
+from requests_mock import Mocker
+from howfairis import Repo
+from howfairis.exceptions import GetDefaultBranchException
+from tests.contracts.repo import Contract
+
+
+def get_mocked_repo():
```
Comment (jspaaks) [inline tests/github/fair_software/doesnotexist/test_repo_no_args.py:8]: I know I started this pattern :), but I was thinking of changing `get_mocked_repo` to `get_repo`. What do you think?

Comment (sverhoeven) [reply tests/github/fair_software/doesnotexist/test_repo_no_args.py:8]: Agree, the repo is not mocked, but the repo on the Internet is faked with the `mocker` fixture.

File: tests/github/fair_software/badge/test_cli_no_args.py (hunk 7 above)
Comment (sverhoeven) [reply tests/github/fair_software/badge/test_cli_no_args.py:72]: Yeah, the indent makes it hard to read.Now that #225 is merged I can implement that.

File: tests/github/fair_software/badge/test_checker_with_reasonable_user_config.py (hunk 6 above)
Comment (sverhoeven) [reply tests/github/fair_software/badge/test_checker_with_reasonable_user_config.py]: Well then I cannot use same name for fixture in test_checker_with_noreason_user_config.py. To me this is a fixture only used in this file. While mocker is used in all tests in the directory.

File: tests/github/fair_software/doesnotexist/test_repo_no_args.py (hunk 8 above)
Comment (sverhoeven) [reply tests/github/fair_software/doesnotexist/test_repo_no_args.py]: I renamed in the files I wrote in this PR, can you create a issue/pr for the others?

File: tests/contracts/cli.py (hunk 2 above)
Comment (sverhoeven) [reply tests/contracts/cli.py]: done

File: tests/github/fair_software/badge/test_checker_with_noreason_user_config.py (hunk 5 above)
Comment (sverhoeven) [reply tests/github/fair_software/badge/test_checker_with_noreason_user_config.py]: lazyness, they have now been copied over

File: tests/github/fair_software/badge/test_cli_no_args.py (hunk 7 above)
Comment (sverhoeven) [reply tests/github/fair_software/badge/test_cli_no_args.py]: done, had to refactor help a bit to get the full path for user files

File: tests/github/fair_software/badge/test_checker_with_noreason_user_config.py (hunk 5 above)
Comment (jspaaks) [reply tests/github/fair_software/badge/test_checker_with_noreason_user_config.py]: spawned #227

File: tests/github/fair_software/badge/test_checker_with_reasonable_user_config.py (hunk 6 above)
Comment (jspaaks) [reply tests/github/fair_software/badge/test_checker_with_reasonable_user_config.py]: ok

File: tests/github/fair_software/badge/test_checker_with_noreason_user_config.py (hunk 1 above)
Comment (jspaaks) [reply tests/github/fair_software/badge/test_checker_with_noreason_user_config.py]: not sure why github isnt showing me the updated code, but ok

File: tests/github/fair_software/doesnotexist/test_repo_no_args.py (hunk 8 above)
Comment (jspaaks) [reply tests/github/fair_software/doesnotexist/test_repo_no_args.py]: done, #228

File: tests/github/fair_software/badge/test_checker_with_noreason_user_config.py (hunk 3 above)
Comment (jspaaks) [reply tests/github/fair_software/badge/test_checker_with_noreason_user_config.py]: we can discuss in later iterations of writing tests, good for now

**Review by jspaaks — APPROVED** [review]
I have a couple more questions and ideas but nothing that should keep us from merging this. Great job! :trophy:

**Comment by sverhoeven** [comment]
Thanks for reviewing,

**Pull request merged**